### PR TITLE
Logging Middleware function added.

### PR DESCRIPTION
**Pull request by rohinb2**
Addresses [#1](https://github.com/HackIllinois/api-commons/issues/1).

- [x] Added middleware that logs the type of request, url, request headers, and request body.

### CODE REVIEW

**Comment by YashoSharma** [comment]
Format this PR properly please

**Comment by YashoSharma** [comment]
Format this PR properly please

**Comment by YashoSharma** [comment]
Format this PR properly please

File: middleware/logging.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+func LoggingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Print(r.Method + " ");
+		fmt.Println(r.URL.Path);
+		fmt.Println("Headers:")
+		for key, value := range r.Header {
+			fmt.Print(key)
+			fmt.Println(value)
```
Comment (justinyangusa) [inline middleware/logging.go]: `fmt.Println` is variadic, so you could combine these two lines with a comma.

File: middleware/logging.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		}
+		fmt.Println("\nBody:")
+		fmt.Println(r.Body)
+		next.ServeHTTP(w, r)
+	})
+}
```
Comment (justinyangusa) [inline middleware/logging.go]: nit: The lack of a newline here is bothering me. In general, you can catch most formatting issues of this kind with `go fmt`.

**Comment by ASankaran** [comment]
Closing logging issues for now. Will reopen later.

**Pull request closed without merging**